### ISSUE TITLE
'NSKeyedUnarchiveFromData' should not be used to for un-archiving and will be removed in a future release

### ISSUE THREAD

**Issue by christoffer-woost** [mixpanel/mixpanel-swift#548]
I recently hit a runtime error in our project.

A bit about my environment.

Minimum deployment target: iOS 14.3
Dependency installed via SPM
Mixpanel version 3.4.0

The error in question
`'NSKeyedUnarchiveFromData' should not be used to for un-archiving and will be removed in a future release`

The offending method
```
static func loadSuperProperties(apiToken: String) -> InternalProperties {
    guard let defaults = UserDefaults(suiteName: MixpanelUserDefaultsKeys.suiteName) else {
        return InternalProperties()
    }
    let prefix = "\(MixpanelUserDefaultsKeys.prefix)-\(apiToken)-"
    guard let superPropertiesData  = defaults.data(forKey: "\(prefix)\(MixpanelUserDefaultsKeys.superProperties)") else {
        return InternalProperties()
    }
    return NSKeyedUnarchiver.unarchiveObject(with: superPropertiesData) as? InternalProperties ?? InternalProperties()
}
```

**Comment by vg-identance** [comment]
Any updates on this?

> 'archivedData(withRootObject:)' was deprecated in iOS 12.0: Use +archivedDataWithRootObject:requiringSecureCoding:error: instead

> 'unarchiveObject(with:)' was deprecated in iOS 12.0: Use +unarchivedObjectOfClass:fromData:error: instead

**Comment by jaredmixpanel** [comment]
Fixed in `v4.1.2`
